### DISCUSSION TITLE
One to one relationship: virtual singleton alias-field as O2M

### DISCUSSION THREAD

**Discussion by Dominic-Marcelino** [directus/directus#16104] [Feature Requests]
Setting up an one-to-one relationship is currently possible by using a Many-to-one relationship with a unique FK.
This can be done in both directions, based on the direction of the relation and whether a delete constraint may be needed.

In case:
* a child-element is always related to a parent
* a child-element should be deleted with it's parent
* A parent element can have zero or one child-elements

the relational FK should belong to into the child-collection.

.

In this case we can only use the o2m alias with the o2m interface to display the child inside the parent-collection. This is missleading, as we can only create one child-element for the parent.

Would it be possible to have either a 'singleton' mode in the O2m interface (hide 'deselect' button, disable 'create' button once an item exists) or even add a virtual O2O alias?

**Comment by azrikahar** [comment]
> a 'singleton' mode in the O2m interface (hide 'deselect' button, disable 'create' button once an item exists) 

Interesting! This does also seem similar to the suggestion here: https://github.com/directus/directus/discussions/12091#discussioncomment-2842936 which also accounts for parent side of things (o2m). 

On one hand having O2M with singleton option might make it clearer what's "under the hood", on the other hand I'm also wondering would a dedicated O2O be easier to understand for non-technical users. 🤔

**Reply by Dominic-Marcelino** [reply]
> On one hand having O2M with singleton option might make it clearer what's "under the hood", on the other hand I'm also wondering would a dedicated O2O be easier to understand for non-technical users. 🤔

As long as directus does not has some kind of auto-setup I think a O2M with singleton would be the way to go. 
This would also allow to easily switch from a O2O to a O2M later by just removing the unique constraint and disabling the singleton option 🤔

As you mentioned a dedicated interface would be easier for non-technical users but I think that would also require directus to support a o2o relation in the setup wizzard. Otherwise I could imagine that non-techinical users probably won't even have a o2o

**Comment by rasco** [comment]
> hide 'deselect' button, disable 'create' button once an item exists

yesss, this! i thought about that too. would be really handy to have these options.
probably could be combined with Conditions.

Also in some cases it could be handy for M2M and M2A as well.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
